CONTRIBUTING.md

```diff
@@ -2,51 +2,57 @@
+
+That said, OpenEBS is an innovation in open-source. You are welcome to contribute in any way you can and all the help provided is very much appreciated. 
+
+- Raise issues to request new functionality, fix documentation or for reporting bugs.
+- Submit changes to improve documentation. 
+- Submit proposals for new features/enhancements.
+- Submit fixes for bugs reported/found. 
+- Submit changes to the source code.
+- Solve existing Issues.
 
 There are just a few simple guidelines that you need to follow before providing your hacks. 
 
 ## Raising Issues
 
-When Raising issues, please specify the following:
-- Setup details (like hyperconverged/dedicated), orchestration engine - kubernetes, docker swarm etc,. 
-- Scenario where the issue was seen to occur
+When raising issues, please specify the following:
+- Setup details (like hyperconverged/dedicated), orchestration engine - kubernetes, docker swarm etc. 
+- Scenario where the issue was seen to occur (With details on how to reproduce it)
 - If the issue is with storage, include maya version, maya osh-status and maya omm-status.
 - Errors and log messages that are thrown by the software
 
-
 ## Submit Change to Improve Documentation
 
-Getting Documentation Right is Hard! Please raise a PR with you proposed changes. 
+Getting documentation right is hard! Refer to this [page](./contribute/CONTRIBUTING-TO-DEVELOPER-DOC.md) page for more information on you could improve the developer documentation by submitting pull requests with appropriate tags. Here's a [list of tags](./contribute/labels-of-issues.md) that could be used. 
+
 
 ## Submit Proposals for New Features
 
-There is always something more that is required, to make it easier to suit your use-cases. Feel free to join the discussion on the features or raise a new PR with your proposed change. 
+There is always something more that is required, to make it easier to suit your use-cases. Feel free to join the discussion on new features or raise a PR with your proposed change. 
 
-## Contributing to Source Code
+## Contributing to Source Code & Bug Fixes
 
-Provide PRs for bug fixes or enhancements.
+Provide PRs with appropriate tags for bug fixes or enhancements to the source code. For a list of tags that could be used, take a look [here]((./contribute/labels-of-issues.md)).
 
+Refer to this [document](./contribute/design/code-structuring.md) for more information on code structuring and some guidelines on the same.
+
+## Solve Existing Issues
+Head over to [issues](www.github.com/openebs/openebs/issues) to find issues where helped is needed from contributors, look over our [list of labels guide](./contribute/labels-of-issues.md) to help you find issues that you can solve faster. **(What guidelines are used here by OpenEBS, need to add the same. - Review needed)**
```

Comment:
where helped is = where help is
contributors, look over = ...contributors. See our

Reply:
Thanks for that. I would like to know if OpenEBS has a guideline or procedure for solving issues. Does someone has to say before taking up an issue, how and where should they state the same. Is it strict to curb redundancies? How to go about it?